Add a note on querying shared columns (#3321)
dotnet/EntityFramework.Docs: entity-framework/core/modeling/inheritance.md

[!code-csharp[Main](../../../samples/core/Modeling/FluentAPI/SharedTPHColumns.cs?name=SharedTPHColumns&highlight=9,13)]

> [!NOTE]

> > Some database providers, such as SQL Server, will not automatically use the discriminator predicate when querying shared columns using a cast. The query `Url = (blob as RssBlog).Url` would also return the `Url` value for the sibling `Blog` rows. To restrict the query to `RssBlog` entities in the hierarchy you need to add a filter manually on the discriminator, such as `Url = blob is RssBlog ? (blob as RssBlog).Url : null`.

Reviewer: ```suggestion
> Relational database providers, such as SQL Server, will not automatically use the discriminator predicate when querying shared columns when using a cast. The query `Url = (blob as RssBlog).Url` would also return the `Url` value for the sibling `Blog` rows. To restrict the query to `RssBlog` entities you need to manually add a filter on the discriminator, such as `Url = blob is RssBlog ? (blob as RssBlog).Url : null`.
```
